buildSrc/src/main/java/com/r3/csde/VNodesHelper.java

```diff
@@ -0,0 +1,288 @@
+     * @return a list of the virtual nodes which have already been created.
+     */
+    private List<VirtualNodeInfoDTO> getExistingNodes () throws CsdeException {
+
+        HttpResponse<JsonNode> response = Unirest.get(pc.baseURL + "/api/v1/virtualnode")
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if(response.getStatus() != HTTP_OK){
+            throw new CsdeException("Failed to get Existing vNodes, response status: "+ response.getStatus());
+        }
+
+        try {
+            return mapper.readValue(response.getBody().toString(), VirtualNodesDTO.class).getVirtualNodes();
+        } catch (Exception e){
+            throw new CsdeException("Failed to get Existing vNodes with exception: " + e);
+        }
+    }
+
+    /**
+     * Creates a vnode on the Corda cluster from the VNode info.
+     * @param vNode represents a virtual node using VNode Class.
+     */
+    private void createVNode(VNode vNode) throws CsdeException {
+
+        pc.out.println("Creating virtual node for "+ vNode.getX500Name());
+
+        // Reads the current CPIFileChecksum value and checks it has been uploaded.
+        String cpiCheckSum = getCpiCheckSum(vNode);
+        if (!checkCpiUploaded(cpiCheckSum)) throw new CsdeException("Cpi " + cpiCheckSum + " not uploaded.");
+
+        // Creates the vnode on Cluster
+        HttpResponse<JsonNode> response = Unirest.post(pc.baseURL + "/api/v1/virtualnode")
+                .body("{ \"request\" : { \"cpiFileChecksum\": \"" + cpiCheckSum + "\", \"x500Name\": \"" + vNode.getX500Name() + "\" } }")
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if (response.getStatus() != HTTP_OK)
+            throw new CsdeException("Creation of virtual node failed with response status: " + response.getStatus());
+    }
+
+    /**
+     * Reads the latest CPI checksums from file.
+     * @param vNode represents a virtual node using VNode Class.
+     */
+    private String getCpiCheckSum(VNode vNode) throws CsdeException {
+
+        try {
+            String file;
+            if (vNode.getServiceX500Name() == null) {
+                file = pc.CPIUploadStatusFName;
+            } else {
+                file = pc.NotaryCPIUploadStatusFName;
+            }
+            FileInputStream in = new FileInputStream(file);
+            CPIFileStatusDTO statusDTO = mapper.readValue(in, CPIFileStatusDTO.class);
+            return statusDTO.getCpiFileChecksum().toString();
+        } catch (Exception e){
+            throw new CsdeException("Failed to read CPI checksum from file, with error: " + e);
+        }
+    }
+
+    private boolean checkCpiUploaded(String cpiCheckSum) throws CsdeException {
+
+        HttpResponse<JsonNode> response = Unirest.get(pc.baseURL + "/api/v1/cpi")
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if(response.getStatus() != HTTP_OK)
```

Comment:
remain consistent with the style ...
I prefer having '{' and '}' even for 1 liner if statements

Reply:
Okay, changed